Content.Server/Mech/Systems/MechSystem.cs

```diff
@@ -93,6 +96,18 @@ private void OnInteractUsing(EntityUid uid, MechComponent component, InteractUsi
         }
     }
 
+    private void OnInsertBattery(EntityUid uid, MechComponent component, EntInsertedIntoContainerMessage args)
+    {
+        if (!TryComp<BatteryComponent>(args.Entity, out var battery))
```

Comment:
check that the container is actually the batter slot container

Reply:
why would you make a powercell mech equipment tho...